### PR TITLE
VPC resource

### PR DESCRIPTION
**Pull request by annakhm**

### CODE REVIEW

File: nsxt/resource_nsxt_vpc.go [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+
+var vpcSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"context":      metadata.GetExtendedSchema(getContextSchema(true, false, false)),
+	"external_ipv4_blocks": {
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc.go:33]: Can validate as policy path

File: nsxt/resource_nsxt_vpc.go [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+			Type:         schema.TypeString,
+			ValidateFunc: validatePolicyPath(),
+			Optional:     true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "VpcServiceProfile",
+		},
+	},
+	"load_balancer_vpc_endpoint": {
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc.go:166]: Can we flatten this? In the end we have MaxItem=1 and a boolean within.

File: nsxt/resource_nsxt_vpc.go [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "LoadBalancerVpcEndpoint",
+			ReflectType:  reflect.TypeOf(model.LoadBalancerVPCEndpoint{}),
+		},
+	},
+	"ip_address_type": {
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc.go:192]: We could drop this entirely, it has only `IPV4` as valid value

File: nsxt/resource_nsxt_vpc.go [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "SubnetProfiles",
+			ReflectType:  reflect.TypeOf(model.SubnetProfiles{}),
+		},
+	},
+	"private_ipv4_blocks": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &schema.Schema{
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc.go:315]: We can validate this as policy path

File: nsxt/resource_nsxt_vpc.go [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+				Type: schema.TypeString,
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "array",
+			SdkFieldName: "PrivateIpv4Blocks",
+		},
+	},
+	"short_id": {
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc.go:325]: This is also computed as NSX sets value to it if user doesn't

Comment (annakhm) [reply nsxt/resource_nsxt_vpc.go:325]: I wonder if this one is also impossible to update..

Comment (ksamoray) [reply nsxt/resource_nsxt_vpc.go:325]: NSX v4.1.2 UI protects it from updating. So I guess it's the same.

File: nsxt/resource_nsxt_vpc.go (hunk 3 above)
Comment (annakhm) [reply nsxt/resource_nsxt_vpc.go:202]: I think we can keep it here, when `ipv6` the enum list will grow automatically as result of resource evolution, and then no manual work will be required
however we might want to skip this in docs

File: nsxt/resource_nsxt_vpc_test.go [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+	"listener_ip":                "230.64.2.1",
+	"enable_dhcp":                "false",
+	"dns_server_ips":             "20.4.12.8",
+}
+
+func TestAccResourceNsxtVpc_basic(t *testing.T) {
+	testResourceName := "nsxt_vpc.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc_test.go:52]: This should be tested for VPC only

File: nsxt/resource_nsxt_vpc.go (hunk 2 above)
Comment (annakhm) [reply nsxt/resource_nsxt_vpc.go:179]: for now the strategy is to follow the api..

**Comment by annakhm** [comment]
waiting for #1286 to rebase and fix the tests

File: nsxt/resource_nsxt_vpc.go (hunk 4 above)
Comment (annakhm) [reply nsxt/resource_nsxt_vpc.go]: This seems to be computed only

File: nsxt/data_source_nsxt_policy_project.go [hunk 7]
```diff
@@ -33,7 +33,14 @@ func dataSourceNsxtPolicyProject() *schema.Resource {
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
-				Optional: true,
+				Computed: true,
+			},
+			"external_ipv4_blocks": {
```
Comment (ksamoray) [inline nsxt/data_source_nsxt_policy_project.go:38]: Change to project data source should be documented

**Pull request merged**